backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

```diff
@@ -1173,83 +1233,55 @@ public Project deleteArtifacts(String projectId, String artifactKey)
       }
     } else {
-
-      // get list of accessible projects
-      @SuppressWarnings("unchecked")
-      List<String> accessibleProjectIds =
-          roleService.getAccessibleResourceIds(
-              null,
-              new CollaboratorUser(authService, currentLoginUserInfo),
-              VisibilityEnum.Visibility.PRIVATE,
-              ModelDBServiceResourceTypes.PROJECT,
-              Collections.EMPTY_LIST);
-      LOGGER.debug(
-          "accessible Project Ids in function getWorkspaceProjectIDs : {}", accessibleProjectIds);
+      Set<String> accessibleAllWorkspaceProjectIds = new HashSet<>();
+      // in personal workspace show projects directly shared
+      if (workspaceName != null
+          && !workspaceName.isEmpty()
+          && workspaceName.equals(authService.getUsernameFromUserInfo(currentLoginUserInfo))) {
+        LOGGER.debug("Workspace and current login user match");
+        accessibleAllWorkspaceProjectIds.addAll(
+            roleService.getSelfDirectlyAllowedResources(
+                ModelDBServiceResourceTypes.PROJECT, ModelDBServiceActions.READ));
+      } else {
+        // get list of accessible projects
+        @SuppressWarnings("unchecked")
+        List<String> accessibleAllProjectIds =
+            roleService.getAccessibleResourceIds(
+                null,
+                new CollaboratorUser(authService, currentLoginUserInfo),
+                VisibilityEnum.Visibility.PRIVATE,
+                ModelDBServiceResourceTypes.PROJECT,
+                Collections.EMPTY_LIST);
+        accessibleAllWorkspaceProjectIds.addAll(accessibleAllProjectIds);
+      }
+      LOGGER.debug("accessibleAllWorkspaceProjectIds : {}", accessibleAllWorkspaceProjectIds);
 
       // resolve workspace
-      WorkspaceDTO workspaceDTO =
-          roleService.getWorkspaceDTOByWorkspaceName(currentLoginUserInfo, workspaceName);
+      Set<String> accessibleProjectIds =
+          ModelDBUtils.filterWorkspaceOnlyAccessibleIds(
+              roleService,
+              accessibleAllWorkspaceProjectIds,
+              workspaceName,
+              currentLoginUserInfo,
+              ModelDBServiceResourceTypes.PROJECT);
 
       List<String> resultProjects = new LinkedList<String>();
       try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
         @SuppressWarnings("unchecked")
-        Query<String> query = session.createQuery(IDS_FILTERED_BY_WORKSPACE);
+        Query<String> query = session.createQuery(NON_DELETED_PROJECT_IDS_BY_IDS);
         query.setParameterList(ModelDBConstants.PROJECT_IDS, accessibleProjectIds);
-        query.setParameter(ModelDBConstants.WORKSPACE, workspaceDTO.getWorkspaceId());
-        query.setParameter(
-            ModelDBConstants.WORKSPACE_TYPE, workspaceDTO.getWorkspaceType().getNumber());
         resultProjects = query.list();
-
-        // in personal workspace show projects directly shared
```

Comment:
How id this being handled after the change? I'm not clear on it.

Reply:
This logic is moved above with workspace https://github.com/VertaAI/modeldb/pull/1795/files/ebc9a94d3c55394424e1ff1c739bf16b477c3986#diff-de14dc78b43342a659f26880706f23c24558f63961a0d39bffad3a86125908e2R1247-R1268

Old logic called two time to database so now i have managed single list based on workspace and call DB to once.

Reply:
That link actually just points me to the top of the page. Not sure why. What are the file and lines that you are referring to?

Reply:
Yes, it is not pointing there but after redirecting go to the file backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java and open the `load diff` it will show you selected block (line 1247-1268)

Reply:
Got it! Checking

Reply:
Thanks! Makes sense. I'm not sure how I missed it before.